frontend/src/components/dashboard/aliases/AliasGenerationButton.tsx

```diff
@@ -0,0 +1,290 @@
+import { useLocalization } from "@fluent/react";
+import {
+  FocusScope,
+  useOverlay,
+  useMenuTrigger,
+  useMenu,
+  DismissButton,
+  mergeProps,
+  useMenuItem,
+  useFocus,
+  useButton,
+} from "react-aria";
+import { OverlayProps } from "@react-aria/overlays";
+import { AriaMenuItemProps } from "@react-aria/menu";
+import {
+  Item,
+  MenuTriggerState,
+  TreeProps,
+  TreeState,
+  useMenuTriggerState,
+  useOverlayTriggerState,
+  useTreeState,
+} from "react-stately";
+import { HTMLAttributes, Key, ReactNode, useRef, useState } from "react";
+import styles from "./AliasGenerationButton.module.scss";
+import plusIcon from "../../../../../static/images/plus-sign-white.svg";
+import arrowHeadIcon from "../../../../../static/images/arrowhead-white.svg";
+import { ProfileData } from "../../../hooks/api/profile";
+import { Button, LinkButton } from "../../Button";
+import { AliasData } from "../../../hooks/api/aliases";
+import { getRuntimeConfig } from "../../../config";
+import { RuntimeData } from "../../../hooks/api/runtimeData";
+import {
+  getPremiumSubscribeLink,
+  isPremiumAvailableInCountry,
+} from "../../../functions/getPlan";
+import { useGaPing } from "../../../hooks/gaPing";
+import { trackPurchaseStart } from "../../../functions/trackPurchase";
+import { AddressPickerModal } from "./AddressPickerModal";
+
+export type Props = {
+  aliases: AliasData[];
+  profile: ProfileData;
+  runtimeData?: RuntimeData;
+  onCreate: (
+    options: { type: "random" } | { type: "custom"; address: string }
+  ) => void;
+};
+
+export const AliasGenerationButton = (props: Props) => {
+  const { l10n } = useLocalization();
```

Comment:
Hmm, I'm not aware of anything off the top of my head, but I'd shy away from implicit globals anyway, because it makes it hard to figure out where something came from if you're new to the codebase, and it would be hard to tell what might be affected by changes to those globals.